src/torchvision_module.py

```diff
@@ -0,0 +1,291 @@
+        use_weight_transform = get_attribute_from_config("use_weight_transform", True)
+        swap_r_and_b = get_attribute_from_config("swap_r_and_b", False)
+        channel_last = get_attribute_from_config("channel_last", False)
+
+        self.preprocessor = Preprocessor(
+            use_weight_transform=use_weight_transform,
+            weights_transform=self.weights.transforms(),
+            input_size=input_size,
+            normalize=(mean_rgb, std_rgb),
+            swap_R_and_B=swap_r_and_b,
+            channel_last=channel_last,
+        )
+
+        self.model.eval()
+
+        self.labels_confidences = get_attribute_from_config(
+            "labels_confidences", dict()
+        )
+        self.default_minimum_confidence = get_attribute_from_config(
+            "default_minimum_confidence", 0
+        )
+
+    async def get_object_point_clouds(
+        self,
+        camera_name: str,
+        *,
+        extra: Optional[Dict[str, Any]] = None,
+        timeout: Optional[float] = None,
+        **kwargs,
+    ) -> List[PointCloudObject]:
+        if not self.properties.implements_get_object_pcd:
+            raise NotImplementedError
+        else:
+            return 1
+
+    async def get_detections(
+        self,
+        image: Union[Image.Image, RawImage],
+        *,
+        extra: Mapping[str, Any],
+        timeout: float,
+    ) -> List[Detection]:
+        if not self.properties.implements_detection:
+            raise NotImplementedError
+        input_tensor = self.preprocessor(image)
+        with torch.no_grad():
+            prediction: Tensor = self.model(input_tensor)[0]
+        labels = [self.weights.meta["categories"][i] for i in prediction["labels"]]
+        scores = prediction["scores"]
+        boxes = prediction["boxes"].to(torch.int64).tolist()
+        res = [
+            Detection(
+                x_min=x_min,
+                y_min=y_min,
+                x_max=x_max,
+                y_max=y_max,
+                confidence=score,
+                class_name=label,
+            )
+            for (x_min, y_min, x_max, y_max), score, label in zip(boxes, scores, labels)
+        ]
+        res = self.filter_output(res)
+        return res
+
+    async def get_classifications(
+        self,
+        image: Union[Image.Image, RawImage],
+        count: int,
+        *,
+        extra: Optional[Mapping[str, Any]] = None,
+        timeout: Optional[float] = None,
+    ) -> List[Classification]:
+        if not self.properties.implements_classification:
+            return NotImplementedError
+
+        input_tensor = self.preprocessor(image)
+        # input_tensor = image
+        # torchvision.utils.save_image(input_tensor, "./input.jpg")
+        # img = decode_image(image)
+        with torch.no_grad():
+            prediction: Tensor = self.model(input_tensor)
+
+        prediction = prediction.squeeze(0)
+        prediction = prediction.softmax(0)
```

Comment:
It's nice that torchvision all follow the same conventions. You wouldn't have been able to assume this in the mlmodel service